Core/clim-basic/geometry/region-set-composition.lisp

```diff
@@ -0,0 +1,192 @@
+(in-package #:climi)
+
+
+;;; utilities
+(defun rectangle-set->polygon-union (rs)
+  (let ((res nil))
+    (map-over-region-set-regions (lambda (r) (push r res)) rs)
+    (make-instance 'standard-region-union :regions res)))
+
+
+;; REGION-UNION
+
+#+ (or) ;; defined in region-composition.lisp
+(defmethod region-union ((a bounding-rectangle) (b bounding-rectangle))
+  (make-instance 'standard-region-union :regions (list a b)))
+
+;;; standard-region-union
+(defmethod region-union ((a standard-region-union) (b standard-region-union))
+  (assert (not (eq b +nowhere+)))
+  (assert (not (eq a +nowhere+)))
+  (make-instance 'standard-region-union
+                 :regions (append (standard-region-set-regions a)
+                                  (standard-region-set-regions b))))
+
+(defmethod region-union ((a standard-region-union) (b bounding-rectangle))
+  (make-instance 'standard-region-union
+                 :regions (cons b (standard-region-set-regions a))))
+
+(defmethod region-union ((b bounding-rectangle) (a standard-region-union))
+  (make-instance 'standard-region-union
+                 :regions (cons b (standard-region-set-regions a))))
+
+;;; standard-rectangle-set
+(defmethod region-union ((a standard-rectangle-set) (b path)) a)
+(defmethod region-union ((b path) (a standard-rectangle-set)) a)
+(defmethod region-union ((a standard-rectangle-set) (b point)) a)
+(defmethod region-union ((b point) (a standard-rectangle-set)) a)
+
+(defmethod region-union ((xs standard-rectangle-set) (ys standard-rectangle-set))
+  (make-standard-rectangle-set
+   (bands-union (standard-rectangle-set-bands xs)
+                (standard-rectangle-set-bands ys))))
+
+(defmethod region-union ((xs standard-rectangle-set) (ys standard-rectangle))
+  (region-union xs (rectangle->standard-rectangle-set ys)))
+
+(defmethod region-union ((xs standard-rectangle) (ys standard-rectangle-set))
+  (region-union (rectangle->standard-rectangle-set xs) ys))
+
+(defmethod region-union ((a standard-rectangle-set) (b polygon))
+  (region-union (rectangle-set->polygon-union a) b))
+
+(defmethod region-union ((a polygon) (b standard-rectangle-set))
+  (region-union a (rectangle-set->polygon-union b)))
+
+;;; standard-region-intersection
+;;; standard-region-difference
+(defmethod region-union ((a standard-region-difference) (b bounding-rectangle))
+  (make-instance 'standard-region-union :regions (list a b)))
+
+(defmethod region-union ((a bounding-rectangle) (b standard-region-difference))
+  (make-instance 'standard-region-union :regions (list a b)))
+
+
+;;; REGION-INTERSECTION
+
+#+ (or) ;; defined in region-composition.lisp
```

Comment:
Can this be removed? Since the method has no region set specializer, I wouldn't expect it to be defined in this file.

Reply:
fixed